src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -251,16 +252,24 @@ type AdaptiveFSharpLspServer(workspaceLoader: IWorkspaceLoader, lspClient: FShar
 
   let builtInCompilerAnalyzers config (file: VolatileFile) (tyRes: ParseAndCheckResults) =
     let filePath = file.FileName
+    let filePathUntag = UMX.untag filePath
     let source = file.Lines
     let version = file.Version
 
+
+    let inline getSourceLine lineNo =
+      (source: ISourceText).GetLineString(lineNo - 1)
+
     let checkUnusedOpens =
       async {
         try
-          let! ct = Async.CancellationToken
+          use progress = new ServerProgressReport(lspClient)
+          do! progress.Begin("Checking unused opens...", message = filePathUntag)
```

Comment:
Adds ProgressReport for users to help identify files which might benefit from disabling.